src/Dialect/ONNX/MLIRDialectBuilder.cpp

```diff
@@ -593,6 +593,27 @@ void SCFBuilder::yield() const { b.create<scf::YieldOp>(loc); }
 // Vector Builder
 //===----------------------------------------------------------------------===//
 
+int64_t VectorBuilder::getMachineVectorLength(const Type &elementType) {
+  unsigned typeBitSize = elementType.getIntOrFloatBitWidth();
+  unsigned simdBitSize;
+  // TODO: use march and mcpu to determine the right size, right now assume
+  // 4*32=128 bits.
+  simdBitSize = 128;
```

Comment:
In order to get the SIMD register size we would need to create a LLVM `TargetTransformInfo` object:
```
DataLayout dl = module.getDataLayout();
TargetTransformInfo TTI(dl);
TypeSize WidestRegister = TTI.getRegisterBitWidth(TargetTransformInfo::RGK_FixedWidthVector);
```

Reply:
Tried your recommendation but it results in errors that I cannot fix at the moment:
``` C++
#include "mlir/IR/BuiltinOps.h"   //hi alex added
#include "mlir/IR/BuiltinTypes.h" // hi alex added
#include "llvm/Analysis/TargetTransformInfo.h" //hi alex added
#include "llvm/IR/DataLayout.h"                // hi alex added

// and in 
int64_t VectorBuilder::getMachineVectorLength(Value vecValue) {

  Operation *op = vecValue.getDefiningOp();
  ModuleOp moduleOp = op->getParentOfType<ModuleOp>();
  DataLayout dl(moduleOp);
  llvm::TargetTransformInfo TTI(dl);

}
```

it results in errors like this
```
[ 36%] Building CXX object src/Dialect/ONNX/CMakeFiles/OMONNXOps.dir/MLIRDialectBuilder.cpp.o
In file included from /workdir/onnx-mlir/src/Dialect/ONNX/MLIRDialectBuilder.cpp:21:
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h: In instantiation of 'const llvm::DataLayout& llvm::TargetTransformInfo::Model<T>::getDataLayout() const [with T = mlir::DataLayout]':
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h:1793:21:   required from here
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h:1794:17: error: 'const class mlir::DataLayout' has no member named 'getDataLayout'; did you mean 'DataLayout'?
 1794 |     return Impl.getDataLayout();
      |            ~~~~~^~~~~~~~~~~~~
      |            DataLayout
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h: In instantiation of 'llvm::InstructionCost llvm::TargetTransformInfo::Model<T>::getGEPCost(llvm::Type*, const llvm::Value*, llvm::ArrayRef<const llvm::Value*>, llvm::TargetTransformInfo::TargetCostKind) [with T = mlir::DataLayout]':
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h:1798:3:   required from here
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h:1801:17: error: 'class mlir::DataLayout' has no member named 'getGEPCost'
 1801 |     return Impl.getGEPCost(PointeeType, Ptr, Operands, CostKind);
      |            ~~~~~^~~~~~~~~~
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h: In instantiation of 'unsigned int llvm::TargetTransformInfo::Model<T>::getInliningThresholdMultiplier() [with T = mlir::DataLayout]':
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h:1803:12:   required from here
/workdir/llvm-project/llvm/include/llvm/Analysis/TargetTransformInfo.h:1804:17: error: 'class mlir::DataLayout' has no member named 'getInliningThresholdMultiplier'
 1804 |     return Impl.getInliningThresholdMultiplier();
      |            ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

when the last statement is removed, `llvm::TargetTransformInfo TTI(dl);` it does not gen errors.

Reply:
ok, we can do this part separately. For this PR lets remove the header files that were added during your experiments and then I am good with it.